tests/gold_tests/post_slow_server/gold/post_slow_server_max_requests_in_0_stderr.gold

```diff
@@ -1,5 +1,5 @@
 ``
 > POST / HTTP/1.1
 ``
-< HTTP/1.1 502 Connection refused
+< HTTP/1.1 502 Broken pipe
```

Comment:
Note to reviewers: 

The `post_slow_server_max_requests_in` was failing with my changes, as it expects a `502 Connection refused`, but gets a `502 Broken pipe`. 

The expectation changed from `502 Broken pipe`->`502 Connection refused` as part of the #9366. Discussed with @bneradt and we believe we should update/revert the expectation to the original. After this PR, certain things seem to be more aligned to how they work pre H2ToOrigin.